src/overlays/actors/ovl_En_Dekunuts/z_en_dekunuts.c

```diff
@@ -71,85 +121,634 @@ static DamageTable D_808BEF64 = {
+                    return;
+                }
+
+                if (this->actor.colChkInfo.damageEffect == 1) {
+                    this->unk_190 = 0x28;
+                    func_800BCB70(&this->actor, 0, 0xFF, 0, 0x28);
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_COMMON_FREEZE);
+                    func_808BE3A8(this);
+                    return;
+                }
+
+                if (this->actor.colChkInfo.damageEffect == 2) {
+                    this->unk_210 = 4.0f;
+                    this->unk_214 = 0.55f;
+                    this->unk_18E = 0;
+                } else if (this->actor.colChkInfo.damageEffect == 4) {
+                    this->unk_210 = 4.0f;
+                    this->unk_214 = 0.55f;
+                    this->unk_18E = 20;
+                    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG,
+                                this->collider.info.bumper.hitPos.x, this->collider.info.bumper.hitPos.y,
+                                this->collider.info.bumper.hitPos.z, 0, 0, 0, 3);
+                } else if (this->actor.colChkInfo.damageEffect == 5) {
+                    this->unk_18E = 32;
+                    this->unk_210 = 4.0f;
+                    this->unk_214 = 0.55f;
+                }
+
+                func_808BE294(this, 1);
+            } else if (this->actor.params == 0) {
+                func_808BDE7C(this);
+            }
+        }
+    } else if ((this->actor.colChkInfo.mass == MASS_IMMOVABLE) && (globalCtx->actorCtx.unk2 != 0) &&
+               (this->actor.xyzDistToPlayerSq < 40000.0f)) {
+        if (this->actor.params == 1) {
+            func_808BDCF0(this);
+        } else if (this->actor.params == 0) {
+            func_808BDE7C(this);
+        } else if (this->actor.colChkInfo.health != 0) {
+            this->actor.colChkInfo.health = 0;
+            func_800BBA88(globalCtx, &this->actor);
+            func_808BE294(this, 0);
+        }
+    }
+}
+
+void EnDekunuts_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDekunuts* this = THIS;
+    s32 pad;
+
+    func_808BE73C(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, this->collider.dim.radius, this->collider.dim.height, 0x1D);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+
+    if (this->collider.base.acFlags & 1) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    if (this->unk_210 > 0.0f) {
+        if (this->unk_18E != 10) {
+            Math_StepToF(&this->unk_210, 0.0f, 0.05f);
+            this->unk_214 = (this->unk_210 + 1.0f) * 0.275f;
+            this->unk_214 = CLAMP_MAX(this->unk_214, 0.55f);
+        } else if ((this->unk_18E == 10) && !Math_StepToF(&this->unk_218, 0.55f, 0.020625f)) {
```

Comment:
```suggestion
        } else if ((this->unk_18E == 10) && !Math_StepToF(&this->unk_218, 0.55f, (33.0f / 1600.0f))) {
```

Reply:
How did you know this one?

Reply:
Just type funny decimals into wolfram alpha and it'll spit out whatever the fraction is (most of the time)